code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm

```diff
@@ -753,280 +753,246 @@
 
 	return ..()
 
-/datum/action/xeno_action/activable/warden_heal/use_ability(atom/A)
-	var/mob/living/carbon/xenomorph/wardenbuff_user = owner
-	if (!istype(wardenbuff_user))
+/datum/action/xeno_action/activable/valkyrie_rage/use_ability(atom/target)
+	var/mob/living/carbon/xenomorph/raging_valkyrie = owner
+	var/mob/living/carbon/xenomorph/buffing_target = target
```

Comment:
this can be a turf, not a mob. and causes very many runtimes. please address this before this is re-tmd